modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteMethod.java

```diff
@@ -122,6 +128,25 @@ public enum IgniteMethod {
         method = Types.lookupMethod(clazz, methodName, argumentTypes);
     }
 
+    /**
+     * Constructor that allows to specify overloaded methods as SQL function implementations.
+     *
+     * @param clazz Class where to lookup method.
+     * @param methodName Method name.
+     * @param overloadedMethod If {@code true} to looks up overloaded methods, otherwise looks up a method w/o parameters.
+     */
+    IgniteMethod(Class<?> clazz, String methodName, boolean overloadedMethod) {
+        if (overloadedMethod) {
+            // Allow calcite to select appropriate method at a call site.
+            this.method = Arrays.stream(IgniteSqlFunctions.class.getMethods())
```

Comment:
that's kinda strange decision: you have particular class as input argument, yet you are using `IgniteSqlFunctions.class`